pageserver/src/remote_storage/storage_sync.rs

```diff
@@ -1062,7 +1084,7 @@ where
                 debug!("Successfully fetched index part for {id}");
                 index_parts.insert(id, index_part);
             }
-            Err(e) => warn!("Failed to fetch index part for {id}: {e:?}"),
```

Comment:
If the timeline has no checkpoints it is ok to have an error there (in case of nothing was uploaded yet). While debugging I spent some time figuring out the reason for this scary backtrace and it was just that :) Since there is no checkpoint for initial timeline every time pageserver starts after init there will be a backtrace